### PR TITLE
Implements Disable Local Auth for Log Analytics Workspaces

### LINKED ISSUE #20091: azurerm_log_analytics_workspace support for Disable Local Auth
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Azure Monitor Logs allows [enforcing a Log Analytics Workspace to only accept Azure AD authentication](https://learn.microsoft.com/en-us/azure/azure-monitor/logs/azure-ad-authentication-logs#disable-local-authentication-for-log-analytics). This feature is useful for environments where regulatory compliance demands central authentication to be in place.

A new flag can be introduced to to control this property configuration.

### New or Affected Resource(s)/Data Source(s)

azurerm_log_analytics_workspace

### Potential Terraform Configuration

```hcl
resource "azurerm_log_analytics_workspace" "laws" {
  name                = "acctest-01"
  location            = azurerm_resource_group.example.location
  ...
  disable_local_auth = true
}
```

### References

https://learn.microsoft.com/en-us/azure/azure-monitor/logs/azure-ad-authentication-logs#disable-local-authentication-for-log-analytics

### PR DESCRIPTION
**Pull request by gusfcarvalho**
Fixes #20091
Signed-off-by: Gustavo <gusfcarvalho@gmail.com>

### CODE REVIEW

**Comment by gusfcarvalho** [comment]
First time contributing to this repo - please let me know if I'm missing something 🙈

**Comment by gusfcarvalho** [comment]
```
make acctests SERVICE='loganalytics' TESTARGS='-run=TestAccLogAnalyticsWorkspace_ToggleDisableLocalAuth' TESTTIMEOUT='60m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/loganalytics -run=TestAccLogAnalyticsWorkspace_ToggleDisableLocalAuth -timeout 60m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccLogAnalyticsWorkspace_ToggleDisableLocalAuth
=== PAUSE TestAccLogAnalyticsWorkspace_ToggleDisableLocalAuth
=== CONT  TestAccLogAnalyticsWorkspace_ToggleDisableLocalAuth
--- PASS: TestAccLogAnalyticsWorkspace_ToggleDisableLocalAuth (462.77s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/loganalytics
```

File: website/docs/r/log_analytics_workspace.html.markdown [hunk 1]
```diff
@@ -39,6 +39,8 @@ The following arguments are supported:
 
 * `allow_resource_only_permissions` - (Optional) Specifies if the log Analytics Workspace allow users accessing to data associated with resources they have permission to view, without permission to workspace. Defaults to `true`.
 
+* `disable_local_auth` - (Optional) Specifices if the log Analytics workspace should enforce authentication using Azure AD. Defaults to `false`.
```
Comment (katbyte) [inline website/docs/r/log_analytics_workspace.html.markdown]: as this is a boolean, it should be consistent with other booleans in the provider (enabled/disabled on the end of the name)
```suggestion
* `local_authentication_disabled` - (Optional) Specifices if the log Analytics workspace should enforce authentication using Azure AD. Defaults to `false`.
```

**Review by katbyte — CHANGES_REQUESTED** [review]
thanks @gusfcarvalho ! overall this looks good with my only comment being around the property name. once that is fixed up i think this will be good to merge 🙂

**Comment by gusfcarvalho** [comment]
Thanks for the review! I’m going for holidays today, I’ll pick this up on Tuesday.

**Comment by gusfcarvalho** [comment]
Updated the PR with the requested name change!

**Comment by gusfcarvalho** [comment]
hi @katbyte ! Can you take a look at it now? Sorry to bother 🙈

**Review by katbyte — APPROVED** [review]
Thanks @gusfcarvalho - LGTM 🦀

**Pull request merged**